RestSharp/RestClient.Async.cs

```diff
@@ -149,7 +149,7 @@ public virtual RestRequestAsyncHandle ExecuteAsyncPost<T>(IRestRequest request,
 
 		private void DeserializeResponse<T>(IRestRequest request, Action<IRestResponse<T>, RestRequestAsyncHandle> callback, IRestResponse response, RestRequestAsyncHandle asyncHandle)
 		{
-			IRestResponse<T> restResponse = response as RestResponse<T>;
+			IRestResponse<T> restResponse = (RestResponse<T>)(response as RestResponse);
```

Comment:
@haf can you remember of this is _the only_ change that is relevant to the original issue?